pkg/token/issuertoken/issuertoken.go

```diff
@@ -0,0 +1,84 @@
+package issuertoken
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/hpe-hcss/errors/pkg/errors"
+)
+
+type TokenResponse struct {
+	TokenType   string `json:"token_type"`
+	ExpiresIn   int    `json:"expires_in"`
+	AccessToken string `json:"access_token"`
+	Scope       string `json:"scope"`
+}
+
+func GenerateIssuerToken(ctx context.Context, issuerURL, clientID, clientSecret string) (string, error) {
+	params := url.Values{}
+	params.Add("client_id", clientID)
+	params.Add("client_secret", clientSecret)
+	params.Add("grant_type", "client_credentials")
+	params.Add("scope", "hpe-tenant")
+
+	url := fmt.Sprintf("%s/v1/token", issuerURL)
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, strings.NewReader(params.Encode()))
+	if err != nil {
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
I think that we may want to add some retry logic here.  iam-lib has a good example:

https://github.com/hpe-hcss/iam-lib/blob/3e5cf1850f38e95bac1962e030789d67b1d42a0b/pkg/identity-client/client.go#L239

	resp, err := c.doRetries(func() (*http.Response, error) {
		return c.httpNewRequest(ctx, http.MethodPost, reqURL, b)
	}, logMessage, 1, targetServiceName, operation)

And:

https://github.com/hpe-hcss/iam-lib/blob/3e5cf1850f38e95bac1962e030789d67b1d42a0b/pkg/identity-client/client.go#L635

      func (c *Client) doRetries(call func() (*http.Response, error), logMessage string, retries int, targetServiceName string,
	      operation string) (*http.Response, error) {
	      var resp *http.Response
	      var err error
	      for {
		      resp, err = call()
		      if err != nil {
			      log.Errorf(logMessage, err)
			      return nil, err
		      }
		      if resp.StatusCode != http.StatusInternalServerError || retries == 0 {
			      break
		      }
		      retries--
		      externalRetryCall.WithLabelValues(targetServiceName, operation).Inc()
	      }
	      return resp, nil
      }



We can modify this to suit our purposes I think.